projects/igniteui-angular/src/lib/grids/state.directive.ts

```diff
@@ -64,23 +77,31 @@ export interface IColumnState {
     searchable: boolean;
 }
 
-const COLUMNS = 'columns';
-const FILTERING = 'filtering';
-const ADVANCED_FILTERING = 'advancedFiltering';
-const SORTING = 'sorting';
-const GROUPBY = 'groupBy';
-const PAGING = 'paging';
-const ROW_SELECTION = 'rowSelection';
-const ROW_PINNING = 'rowPinning';
-const PINNING_CONFIG = 'pinningConfig';
-const CELL_SELECTION = 'cellSelection';
-const COLUMN_SELECTION = 'columnSelection';
+export type GridFeatures = keyof IGridStateOptions;
+
+export interface Feature {
```

Comment:
For the public exports I'd keep to prefixing interfaces for the state directive with `IGridState<X>` to avoid collisions with anything else that might be named "Feature" :)

Reply:
On second thought, should this be exported at all?

Reply:
No need to be exported :)